package/MDAnalysis/coordinates/GSD.py

```diff
@@ -0,0 +1,129 @@
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+"""GSD trajectory reader  --- :mod:`MDAnalysis.coordinates.GSD`
+============================================================
+
+Class to read the GSD trajectory, output of HOOMD-blue. The GSD format specifies
+both the topology and the trajectory of the particles in the simulation. The
+topology is read by the GSDTopology class.
+
+The GSD format was developed having in mind the possibility of changing number
+of particles, particle types, particle identities and changing topology.
+Currently this class has limited functionality, due to the fact that the number
+of particles and the topology are kept fixed in most MD simulations. The user
+will get an error only if at any time step the number of particles is detected
+to be different to the one that was set at the first time step. No check on
+changes in particle identity or topology is currently implemented.
+
+Classes
+-------
+
+.. autoclass:: GSDReader
+   :inherited-members:
+
+"""
+from __future__ import absolute_import, division
+
+import numpy as np
+from . import base
+try :
+    import gsd.hoomd
+except :
+    pass
+
+class GSDReader(base.ReaderBase):
+    """Reader for the GSD format.
+
+    """
+    format = 'GSD'
+    units = {'time': None, 'length': None}
+
+    def __init__(self, filename, **kwargs):
+        """
+        Parameters
+        ----------
+        filename : str
+            trajectory filename
+        **kwargs : dict
+            General reader arguments.
+
+
+        .. versionadded:: 0.17.0
+        """
+        super(GSDReader, self).__init__(filename, **kwargs)
+        self.filename = filename
+        self.open_trajectory()
+        self.n_atoms = self._file[0].particles.N
+        self.ts = self._Timestep(self.n_atoms, **self._ts_kwargs)
+        self._read_next_timestep()
+
+    def open_trajectory(self) :
+        """opens the trajectory file using gsd.hoomd module"""
+        self._frame = -1
+        self._file = gsd.hoomd.open(self.filename,mode='rb')
```

Comment:
here it would be nice to have a good error message for the user. In general, in the init function you should make a test if the gsd module was available and then raise an error if it wasn't found. That is better then the error message this line will throw "no module gsd"